cogs/tags.py

```diff
@@ -107,115 +170,338 @@ async def all(self, ctx: commands.Context):
+            for record in records:
+                pager.add_line(line=record["name"])
 
-        for page in pager.pages:
-            await asyncio.sleep(1)
-            await ctx.author.send(page)
+            for page in pager.pages:
+                await asyncio.sleep(1)
+                await ctx.author.send(page)
+
+        asyncio.create_task(send_tags())
 
-        await ctx.send("Tags sent in DMs.")
+        await ctx.send("Tags are being sent in DMs.")
 
     @tag.command()
     @is_engineer_check()
     async def edit(self, ctx, name: lambda inp: inp.lower(), *, text: str):
         """Edit a tag"""
+        name = await commands.clean_content().convert(ctx=ctx, argument=name)
         text = await commands.clean_content().convert(ctx=ctx, argument=text)
 
         tag = await self.bot.db.get_tag(guild_id=ctx.guild.id, name=name)
 
         if tag is None:
             await ctx.message.delete(delay=10.0)
-            message = await ctx.send('Could not find a tag with that name.')
+            message = await ctx.send("Could not find a tag with that name.")
             return await message.delete(delay=10.0)
 
         if tag.creator_id != ctx.author.id:
             if not is_admin(ctx.author):
-                return await ctx.send('You don\'t have permission to do that.')
+                return await ctx.send("You don't have permission to do that.")
+
+        if is_staff(ctx.author):
+            await tag.update(text=text)
+
+            await self.log_channel.send(
+                embed=discord.Embed(
+                    title="Tag Updated",
+                    color=discord.Color.green(),
+                    description=f"**Before**: {tag.text}\n\n**After**: {text}",
+                ).set_footer(
+                    text=f"Updated By: {ctx.author}", icon_url=ctx.author.avatar_url
+                )
+            )
+
+            return await ctx.send("You have successfully edited your tag.")
+
+        log = await self.log_channel.send(
+            embed=discord.Embed(
+                title="Tag Update Request",
+                timestamp=datetime.utcnow(),
+                color=discord.Color.blurple(),
+                description=f"**After**: {text}",
+            )
+            .add_field(name="Before", value=tag.text, inline=False)
+            .add_field(name="Name", value=name)
+            .add_field(name="Creator ID", value=ctx.author.id)
+            .set_author(name=ctx.author, icon_url=ctx.author.avatar_url),
+        )
+
+        for emoji in EMOJIS:
+            await log.add_reaction(emoji)
 
-        await tag.update(text=text)
-        await ctx.send('You have successfully edited your tag.')
+        return await ctx.reply("Tag update request submitted.")
 
     @tag.command()
     @is_engineer_check()
     async def delete(self, ctx, *, name: lambda inp: inp.lower()):
         """Delete a tag."""
+        name = await commands.clean_content().convert(ctx=ctx, argument=name)
```

Comment:
I think `name: commands.clean_content` then `name = name.lower()` would be clearer than invoking the converter for clean content.